File: app/src/main/java/com/bmn/bookfinder/adapters/CreateAccountViewPagerAdapter.java
package com.bmn.bookfinder.adapters;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import java.util.ArrayList;
import java.util.List;

public class CreateAccountViewPagerAdapter extends FragmentStatePagerAdapter {

    private final List<Fragment> mFragmentList = new ArrayList<>();


    public CreateAccountViewPagerAdapter(FragmentManager fragment) {
        super(fragment, FragmentStatePagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        return mFragmentList.get(position);
    }

    public void addFragment(Fragment fragment) {
        mFragmentList.add(fragment);
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return "Login";
            case 1:
                return "Sign up";
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return mFragmentList.size();
    }
}

File: app/src/main/java/com/bmn/bookfinder/adapters/CreateAccountViewPagerAdapter.kt
package com.bmn.bookfinder.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import java.util.*

class CreateAccountViewPagerAdapter(fragment: FragmentManager?) : FragmentStatePagerAdapter(
    fragment!!, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT
) {
    private val mFragmentList: MutableList<Fragment> = ArrayList()
    override fun getItem(position: Int): Fragment {
        return mFragmentList[position]
    }

    fun addFragment(fragment: Fragment) {
        mFragmentList.add(fragment)
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return when (position) {
            0 -> "Login"
            1 -> "Sign up"
            else -> null
        }
    }

    override fun getCount(): Int {
        return mFragmentList.size
    }
}